### PR TITLE
ttl: refractor ttl pkg into multiple sub packages

### LINKED ISSUE #39490: refractor ttl package into multiple sub packages
## Enhancement

Writing all logics in the same package is not suggested. As we'll split the manager / worker into a seperated package, the util packages should also be seperated to avoid the reverse dependency from `/ttl/ttlworker` to `/ttl`.

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What problem does this PR solve?

Issue Number: close #39490

Problem Summary:

Split the util functions inside ttl into multiple standalone packages.

### What is changed and how it works?

Just move the files, and change some headers. Nothing new.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

**Review by lcwangchao — APPROVED** [review]

File: ttl/cache/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        "@com_github_pingcap_errors//:errors",
+    ],
+)
+
+go_test(
+    name = "cache_test",
+    srcs = [
+        "main_test.go",
+        "table_test.go",
+    ],
```
Comment (hawkingrei) [inline ttl/cache/BUILD.bazel:25]: Please ```flaky = True,``` and the other need to do too.

Comment (YangKeao) [reply ttl/cache/BUILD.bazel:25]: OK.

Is it still needed even when the test is not flaky?

Comment (YangKeao) [reply ttl/cache/BUILD.bazel:25]: Added.

At least, I understand it's better than not.

Comment (hawkingrei) [reply ttl/cache/BUILD.bazel:25]: Yes, It is defensive

**Comment by YangKeao** [comment]
@hawkingrei Could you help me to merge it again? I run the `make bazel_prepare` and pushed a commit to reorder the position `flaky = True` (keep consistent with the generated position).

**Pull request merged**